analyzers/src/SonarAnalyzer.Common/Rules/Hotspots/DoNotHardcodeCredentialsBase.cs

```diff
@@ -27,30 +28,14 @@
 
 namespace SonarAnalyzer.Rules
 {
-    public abstract class DoNotHardcodeCredentialsBase<TSyntaxKind> : ParametrizedDiagnosticAnalyzer
-        where TSyntaxKind : struct
+    public abstract class DoNotHardcodeCredentialsBase : ParametrizedDiagnosticAnalyzer
     {
-        protected const char CredentialSeparator = ';';
-        private const string DiagnosticId = "S2068";
-        private const string MessageFormat = "{0}";
-        private const string MessageHardcodedPassword = "Please review this hard-coded password.";
-        private const string MessageFormatCredential = @"""{0}"" detected here, make sure this is not a hard-coded credential.";
-        private const string MessageUriUserInfo = "Review this hard-coded URI, which may contain a credential.";
         private const string DefaultCredentialWords = "password, passwd, pwd, passphrase";
+        private static readonly ConcurrentDictionary<string, Regex> PasswordValuePattern = new();
+        protected static readonly Regex ValidCredentialPattern = new(@"^(\?|:\w+|\{\d+[^}]*\}|""|')$", RegexOptions.IgnoreCase | RegexOptions.Compiled);
```

Comment:
I've opened https://github.com/SonarSource/sonar-dotnet/issues/8314 to deal with the timeouts. It should be a small easy change.